### DISCUSSION TITLE
Hidden Fields to pass values to the backend / Formsnap

### DISCUSSION THREAD

**Discussion by eftichis0202** [svecosystem/formsnap#166] [Q&A]
Hi all,

hope this finds you well. I am trying to figure out how to pass values to the back-end using superforms and formsnap. For some reason i can't get it to work..

For example:
`<input` type="hidden" name="companyid" value={$currentCompany.id} />

inside the form field, will not add the value to companyid for some reason.
How do i need to do this using form fields.

The schema.ts:

```
export const addVorhabenSchema = z.object({
	vorhaben_name: z.string().min(3, 'Bitte mehr als 3 Zeichen eingeben.'),
	companyid: z.string().min(1)
});
```

Thanks for your help,

Felix

**Comment by ambergristle** [comment]
this seems more like a svelte than a formsnap question. i believe that in order for the input to update form state (or a local variable or store), its value must either be bound to it, or the on:change cb must be used:
```
<input 
  name="id" 
  bind:value={$currentCompanyId} 
/>
<!-- or -->
<input 
  name="id" 
  value={$currentCompanyId} 
  on:change={(event) => { $currentCompanyId = event.target.value }} 
/>
```
